ml4gw/transforms/decimator.py

```diff
@@ -0,0 +1,195 @@
+import torch
+
+
+class Decimator(torch.nn.Module):
+    """
+    Downsample/decimate timeseries according to a schedule.
+
+    The schedule is a tensor of shape (N, 3), where each row is:
+    [start_time, end_time, target_sample_rate]
+
+    Example default schedule (seconds, Hz):
+    [[0, 40, 256],   # first 40s at 256 Hz
+     [40, 58, 512],  # next 18s at 512 Hz
+     [58, 60, 2048]] # last 2s at full 2048 Hz
+
+    Strain: Tensor of shape (B, C, T), where
+    B=batch size, C=channels, T=time.
+
+    Functions:
+        build_variable_indices(sample_rate, schedule, device)
+            Compute the sample indices to keep from the full strain.
+        split_by_schedule(strain, schedule)
+            Split a strain into segments according to the schedule.
+    """
+
+    def __init__(
+        self,
+        sample_rate: int = None,
+        schedule: torch.Tensor = None,
+        device=None,
+    ):
+        super().__init__()
+        self.sample_rate = sample_rate
+        self.schedule = schedule
+        self.device = device
+
+        if sample_rate is not None and schedule is not None:
+            self._validate_inputs(sample_rate, schedule)
+            self.idx = self.build_variable_indices(
```

Comment:
When you calculate `idx` here, you can store it as a buffer with `register_buffer`; there are some examples in the other transforms. This will allow you to not need to specify the device anywhere.